src/uct/tcp/tcp_cm.c

```diff
@@ -148,6 +148,42 @@ static void uct_tcp_cm_trace_conn_pkt(const uct_tcp_ep_t *ep,
             uct_tcp_ep_get_cm_id(ep));
 }
 
+ucs_status_t uct_tcp_cm_send_event_pending_cb(uct_pending_req_t *self)
+{
+    uct_tcp_ep_pending_req_t *req =
+            ucs_derived_of(self, uct_tcp_ep_pending_req_t);
+    ucs_status_t status;
+
+    status = uct_tcp_cm_send_event(req->ep, req->cm.event, req->cm.log_error);
+    ucs_free(req);
```

Comment:
check status before releasing. release only if status is OK

Reply:
if you return UCS_ERR_NO_MEMORY from pending, it will remain on the pending queue
https://github.com/openucx/ucx/blob/1287d3e262800d91e3bf84e2687e0e2164766174/src/uct/base/uct_iface.h#L583

Reply:
got it, fixed